src/main/java/SYSC4806/Application.java

```diff
@@ -34,32 +34,86 @@ public CommandLineRunner demo(CategoryRepository categoryRepository, CourseRepos
             Category category2 = new Category("Software Architecture");
             Category category3 = new Category("Logic Algorithm");
             LearningOutcome learningOutcome = new LearningOutcome("Web Application Coding", category);
+            LearningOutcome learningOutcome2 = new LearningOutcome("Design Software System", category2);
+            LearningOutcome learningOutcome3 = new LearningOutcome("Logic Gates", category3);
             Course course = new Course(4, "Software Engineering Lab", "4806");
             Course course2 = new Course(3, "Real-Time Concurrent System", "3303");
             Course course3 = new Course(3, "Operating Systems", "4001");
+            Course course4 = new Course(2, "Circuits and Signals", "2501");
+            Course course5 = new Course(2, "Switching Circuits", "2607");
+            Course course6 = new Course(1, "Introduction to Software", "1005");
+            Course course7 = new Course(1, "Introduction to Engineering", "1010");
             Program program = new Program("Software Engineering");
             Program program2 = new Program("Civil Engineering");
             Program program3 = new Program("Computer System Engineering");
+            Program program4 = new Program("Electrical Engineering");
 
             category.addLO(learningOutcome);
+            category2.addLO(learningOutcome2);
+            category3.addLO(learningOutcome3);
             learningOutcome.addCourse(course);
+            learningOutcome.addCourse(course2);
+            learningOutcome.addCourse(course3);
+            learningOutcome.addCourse(course6);
+            learningOutcome2.addCourse(course7);
+            learningOutcome3.addCourse(course4);
+            learningOutcome3.addCourse(course5);
 
             course.addLO(learningOutcome);
             course.addProgram(program);
+            course2.addLO(learningOutcome);
+            course2.addProgram(program);
+            course3.addLO(learningOutcome);
+            course3.addProgram(program);
+
+            // Problems occurred after adding these.
```

Comment:
I ran this and for me, it only fails on the following commented lines: 

```
           // Problems occurred after adding these.
            course4.addLO(learningOutcome3);
            course4.addProgram(program4);
            course4.addProgram(program3);
//            course4.addProgram(program);
            course5.addLO(learningOutcome3);
            course5.addProgram(program4);
            course5.addProgram(program3);
//            course5.addProgram(program);
            course6.addLO(learningOutcome);
            course6.addProgram(program);
//            course6.addProgram(program3);
            course7.addLO(learningOutcome2);
            course7.addProgram(program2);
```

Reply:
I have a fix for this coming up, there were some remaining issues for mapping that I'm just fixing and testing now.